### PR TITLE
Rfc: use block indent for multi-lined imports

### LINKED ISSUE #1784: `use`s aren't block-formatted
With the `src/lib.rs`

```
use self::unicode_normalization::{UnicodeNormalization, Recompositions};
```

and the `rustfmt.toml`

```
max_width = 70
reorder_imports = true
reorder_imports_in_group = true
tab_spaces = 2
struct_variant_width = 0
write_mode = "Overwrite"
```

`cargo fmt` leaves:

```
use self::unicode_normalization::{UnicodeNormalization,
                                  Recompositions};
```

### PR DESCRIPTION
**Pull request by topecongiro**
Closes #1784.

### CODE REVIEW

**Comment by nrc** [comment]
Hmm, it turns out we didn't agree on a style for this (https://github.com/rust-lang-nursery/fmt-rfcs/issues/24), so I think we should not change the default for now. Could we put this behind an option for now?

**Comment by topecongiro** [comment]
Looks like using multiple imports are not welcomed, so I will change this PR to support A, C and D.

**Comment by topecongiro** [comment]
@nrc Do you think we could set `reorder_imported_names` to `true` as default? I think this is agreed in https://github.com/rust-lang-nursery/fmt-rfcs/issues/24. If so, we need to implement better sorting.

**Comment by nrc** [comment]
> Do you think we could set reorder_imported_names to true as default?

Yes, I think that would be great (once the sorting matches the spec).

**Comment by topecongiro** [comment]
Updated & rebased.

File: Configurations.md [hunk 1]
```diff
@@ -979,6 +979,85 @@ match lorem {
...
+use foo::{aaa, bbb, ccc,
+          ddd, eee, fff};
+```
+
+#### `"Horizontal"`
+
+```rust
+use foo::{xxx, yyy, zzz};
+
+use foo::{aaa, bbb, ccc, ddd, eee, fff};
```
Comment (nrc) [inline Configurations.md]: What happens when this gets too long to fit on one line?

Comment (topecongiro) [inline Configurations.md]: When `Horizontal` layout is spcecified, rustfmt forces horitonzal layout and puts every thing on one line even if it exceeds max width.

**Comment by topecongiro** [comment]
Added note to Configurations.md about using `Horizontal` option.

**Pull request merged**